### PR TITLE
Support "async function" in `modifyFunction`

### PR DESCRIPTION
**Pull request by tlively**
This is required to use `modifyFunction` in PthreadFS.

### CODE REVIEW

**Comment by tlively** [comment]
Ideas on how to test this would be very welcome. Relatedly, how should we pull in PthreadFS from https://github.com/rstz/emscripten-pthreadfs to add a test to make sure it continues to at least parse correctly?

**Comment by sbc100** [comment]
> Ideas on how to test this would be very welcome. Relatedly, how should we pull in PthreadFS from https://github.com/rstz/emscripten-pthreadfs to add a test to make sure it continues to at least parse correctly?

Do we want to take on the responsibility?  Isn't the main point of not have that upstream is that was don't need to maintain it over time?

File: src/parseTools.js [hunk 1]
```diff
@@ -931,32 +931,35 @@ function makeRetainedCompilerSettings() {
...
-  let match = text.match(/^\s*function\s+([^(]*)?\s*\(([^)]*)\)/);
+  let match = text.match(/^\s*(async\s+)?function\s+([^(]*)?\s*\(([^)]*)\)/);
+  let async_;
   let names;
   let args;
   let rest;
   if (match) {
-    name = match[1];
-    args = match[2];
+    async_ = match[1] || '';
```
Comment (sbc100) [inline src/parseTools.js:944]: how about calling this `isAsync`?

**Review by sbc100 — APPROVED** [review]

**Comment by tlively** [comment]
> > Ideas on how to test this would be very welcome. Relatedly, how should we pull in PthreadFS from https://github.com/rstz/emscripten-pthreadfs to add a test to make sure it continues to at least parse correctly?
> 
> Do we want to take on the responsibility? Isn't the main point of not have that upstream is that was don't need to maintain it over time?

Yeah I ideally we wouldn't have to maintain it, but in the short term it will be less work to make sure we don't break it than to have to spend multiple hours debugging it when we do break it.

File: src/parseTools.js (hunk 1 above)
Comment (tlively) [reply src/parseTools.js:944]: That doesn't seem quite right because it's not a boolean, it's a string.

Comment (sbc100) [reply src/parseTools.js:944]: oh.. i see..

**Comment by sbc100** [comment]
> > > Ideas on how to test this would be very welcome. Relatedly, how should we pull in PthreadFS from https://github.com/rstz/emscripten-pthreadfs to add a test to make sure it continues to at least parse correctly?
> > 
> > 
> > Do we want to take on the responsibility? Isn't the main point of not have that upstream is that was don't need to maintain it over time?
> 
> Yeah I ideally we wouldn't have to maintain it, but in the short term it will be less work to make sure we don't break it than to have to spend multiple hours debugging it when we do break it.

We don't have any present do pulling in third party code like that as part of our testing.

If we want to take on the testing burden perhaps we should just move it upstream.

**Pull request merged**